core/server/api/shared/headers.js

```diff
@@ -0,0 +1,50 @@
+const INVALIDATE_ALL = '/*';
+
+const cacheInvalidate = (result, options = {}) => {
+    let value = options.value;
+
+    return {
+        'X-Cache-Invalidate': value || INVALIDATE_ALL
+    };
+};
+
+const disposition = {
+    csv(result, options = {}) {
+        return {
+            'Content-Disposition': options.value,
+            'Content-Type': 'text/csv'
+        };
+    },
+
+    json(result, options = {}) {
+        return {
+            'Content-Disposition': options.value,
+            'Content-Type': 'application/json',
+            'Content-Length': JSON.stringify(result).length
+        };
+    },
+
+    yaml(result, options = {}) {
+        return {
+            'Content-Disposition': options.value,
+            'Content-Type': 'application/yaml',
+            'Content-Length': JSON.stringify(result).length
+        };
+    }
+};
```

Comment:
Maybe `disposition` can be a fn as well like `cacheInvalidate(result, options)` above, and then use `switch-case` over `options.type` to return the correct value? All 3 disposition methods take same params right now, so its redundant, plus it will be more consistent. Main reason was also to make below call more readable, so 
> Object.assign(headers, disposition[config.disposition.type](result, config.disposition));
becomes
> Object.assign(headers, disposition(result, config. disposition));

Reply:
Hmm we could that yeah, but no priority in my opinion? It's IMO not unreadable.

Reply:
Yeah def not a readability problem as such, just an optimization, could be done later 👍